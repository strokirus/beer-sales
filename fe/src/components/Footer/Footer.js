import React from 'react';
import { func } from 'prop-types';

import {
  MainFooter,
} from './styles';

/**
 * Component to Footer
 * @param toggleTheme Function to toggle theme
*/
const Footer = ({ toggleTheme }) => (
  <MainFooter>
    <p
      className="about"
    >
      Created by
      <a
        href="//diasalexandre.com"
        target="_blank"
        rel="noopener noreferrer"
      >
        Alexandre Dias
      </a>
    </p>
    <p
      className="toggle"
    >
      <button
        type="button"
        onClick={toggleTheme}
      >
        Toggle theme
      </button>
    </p>
  </MainFooter>
);

Footer.propTypes = {
  toggleTheme: func,
};

Footer.defaultProps = {
  toggleTheme: () => {},
};

export default Footer;
