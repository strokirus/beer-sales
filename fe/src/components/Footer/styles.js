import styled from 'styled-components';

export const MainFooter = styled.footer`
  position: absolute;
  bottom: 0;
  left: 0;
  padding: 20px 0;
  display: flex;
  flex-start: end;
  width: 100%;
  justify-content: center;

  > .about {
    flex: 0 0 48%;
    width: 48%;
    text-align: center;

    > a {
      margin-left: 5px;
    }
  }

  > .toggle {
    flex: 0 0 48%;
    width: 48%;
    text-align: center;
  }
`;

export default {
  MainFooter,
};
