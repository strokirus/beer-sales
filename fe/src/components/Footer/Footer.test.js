import React from 'react';
import renderer from 'react-test-renderer';
import {
  mount,
  configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Footer from './Footer';

configure({ adapter: new Adapter() });

describe('(Component) Footer', () => {
  it('should render an Footer', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call toggleTheme event', () => {
    const onToggleThemeEventFn = jest.fn();

    const wrapper = mount(
      <Footer
        toggleTheme={onToggleThemeEventFn}
      />,
    );
    expect(onToggleThemeEventFn).toHaveBeenCalledTimes(0);

    wrapper
      .find('button[type="button"]')
      .first()
      .simulate('click');

    expect(onToggleThemeEventFn).toHaveBeenCalledTimes(1);
  });
});
