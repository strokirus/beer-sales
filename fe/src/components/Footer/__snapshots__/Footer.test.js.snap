// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Footer should render an Footer 1`] = `
<footer
  className="sc-AxjAm hAmNEn"
>
  <p
    className="about"
  >
    Created by
    <a
      href="//diasalexandre.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Alexandre Dias
    </a>
  </p>
  <p
    className="toggle"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Toggle theme
    </button>
  </p>
</footer>
`;
